test/jdk/java/nio/channels/unixdomain/NonBlockingAccept.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8245194
+ * @run main/othervm NonBlockingAccept
+ */
+
+import java.net.StandardProtocolFamily;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+
+public class NonBlockingAccept {
+
+    static boolean supported() {
+        try {
+            SocketChannel.open(StandardProtocolFamily.UNIX).close();
+        } catch (UnsupportedOperationException e) {
+            return false;
+        } catch (Exception e) {
+            return true;
+        }
+        return true;
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        if (!supported()) {
+            return;
+        }
+
+        try (ServerSocketChannel serverSocketChannel =
+                                 ServerSocketChannel.open(StandardProtocolFamily.UNIX)) {
+            //non blocking mode
+            serverSocketChannel.configureBlocking(false);
+            serverSocketChannel.bind(null);
+            SocketChannel socketChannel = serverSocketChannel.accept();
+            System.out.println("The socketChannel is : expected Null " + socketChannel);
+            // exception could be thrown otherwise
+        }
```

Comment:
Do we have/need a test that goes the whole way down? Register for OP_CONNECT with a selector and wait for the connect to finish?

Reply:
Yes, the IOExchanges test covers many different scenarios around that